rules/modular-monoliths/rule.md

```diff
@@ -1,38 +1,59 @@
 
 <!--endintro-->
 
-### Key aspects
+> The Modular Monolith architecture is the “goldilocks” approach that combines the modularity of microservices with the simplicity of traditional Monoliths
+>
+> * Steve “Ardalis” Smith
+
+## Modular Monolith characteristics
 
-1. **Monolithic Architecture** - Traditionally, a monolithic application is a single-tiered software application where different components are combined into a single project. This means all components of the application are interconnected and interdependent
+![Figure: Modular Monolith architecture](modular-monolith.jpg)
 
-2. **Modularity** - In a Modular Monolith, the application is still a monolith, but it's structured in a way that components or modules within it are highly modular. This modularity refers to dividing the application into distinct modules, each responsible for a specific piece of functionality. Eg. Invoicing
-These modules are designed to be loosely coupled, meaning they interact with each other through well-defined interfaces and are largely independent in terms of development and deployment
+* Single Host/Process
+* Single Deployment
+* Loosely coupled modules that each have their own
+  * Domain
+  * Application
+  * Infrastructure
+  * Presentation (API or UI)
+* Each module represents a business capability or domain
+* Each module should be as highly cohesive and loosely coupled with other modules
+* Each module manages it's on data and persistence
 
-3. **Use Cases* - It's often used in scenarios where an application is expected to be complex but not so large or varied that it requires a distributed architecture. It's also a good fit for teams transitioning from a monolithic architecture to microservices, as it offers a middle ground
+### ✅ Advantages
 
-#### ✅ Advantages
+* **Simplicity in Deployment** - Since it's a monolith, the deployment is typically simpler than distributed systems like microservices
+* **Ease of Development** - Developers can work on separate modules without significantly affecting other parts of the application
+* **Performance** - Inter-module communication is often faster and more reliable than inter-service communication in distributed architectures
 
-- **Simplicity in Deployment**: Since it's a monolith, the deployment is typically simpler than distributed systems like microservices
-- **Ease of Development**: Developers can work on separate modules without significantly affecting other parts of the application
-- **Performance**: Inter-module communication is often faster and more reliable than inter-service communication in distributed architectures
+### ❌ Challenges
 
-#### ❌ Challenges
+* **Scalability** - While more scalable than a traditional monolith, it may not scale as effectively as microservices
+* **Modular Discipline** - Maintaining strict modularity can be challenging as the application grows and evolves
```

Comment:
This adherence to the strict rules of modular monoliths will be up to the **team** and/or **code reviewers** to enforce, and it gets harder to enforce as the app grows. This is the one downside of MM/VSA vs the layered Clean Architecture. The layering automatically enforces the direction of dependencies